solutions/standard/variables.tf

```diff
@@ -0,0 +1,86 @@
+# Common
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API Key"
+  sensitive   = true
+}
+
+# Namespace
+variable "use_existing_resource_group" {
+  type        = bool
+  description = "Indicates whether to use an existing resource group. If set to 'false', a new resource group will be created."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of the resource group where resources will be provisioned. This can either be an existing resource group or a new one."
+  default     = "icr-namespace"
+}
+
+variable "region" {
+  type        = string
+  description = "The IBM Cloud region where the container registry namespace and retentation policy will be created."
```

Comment:
It seems your only using `region` in the provider config. Meaning its going to apply to all resources in the DA. Since the namespace resources doesn't take in any region value, I suspect it takes it from the provider config? So what happens if a user want to create a global namespace? Does region support taking in the string `global`? If so, I suspect you will want to use a provider alias just for namespace creation.

Reply:
Yes, using global as a region works

I have added a new provider block with `namespace` alias and also renamed `region` variable to `namespace_region`

This block will be used for resource group and to get a iam token
```
provider "ibm" {
  ibmcloud_api_key = var.ibmcloud_api_key
  region           = "us-south"
}
```

This block will be used to create namespace and set retentation policy
```
provider "ibm" {
  alias            = "namespace"
  ibmcloud_api_key = var.ibmcloud_api_key
  region           = var.namespace_region
}
```